gcloud-java-examples/src/main/java/com/google/gcloud/examples/StorageExample.java

```diff
@@ -213,8 +214,12 @@ String parse(String... args) {
     public void run(Storage storage, String bucketName) {
       if (bucketName == null) {
         // list buckets
-        for (BucketInfo b : storage.list()) {
-          System.out.println(b);
+        Page<BucketInfo> bucketPage = storage.list();
```

Comment:
Yes, the first page is never null. At first I wrote it with a `do - while` and also added a `hasNextPage` method to `Page` to avoid checking for null but I did not want to put a (very rarely used) `do - while` loop in our snippets.

If you think that would be better I can update the snippet (and the example) to use `do - while`, I have no strong opinions here.